src/cli_plugin/install/settings.js

```diff
@@ -0,0 +1,48 @@
+import expiry from 'expiry-js';
+import { intersection } from 'lodash';
+import { resolve } from 'path';
+import { arch, platform } from 'os';
+
+function generateUrls(settings) {
+  const { version, plugin } = settings;
+  return [
+    plugin,
+    `https://download.elastic.co/packs/${plugin}/${plugin}-${version}.zip`
+  ];
+}
+
+export function parseMilliseconds(val) {
+  let result;
+
+  try {
+    let timeVal = expiry(val);
+    result = timeVal.asMilliseconds();
+  } catch (ex) {
+    result = 0;
+  }
+
+  return result;
+};
+
+export function parse(command, options, kbnPackage) {
+  const settings = {
+    timeout: options.timeout ? options.timeout : 0,
+    quiet: options.quiet ? options.quiet : false,
+    silent: options.silent ? options.silent : false,
+    config: options.config ? options.config : '',
+    plugin: command,
+    version: kbnPackage.version,
+    pluginDir: options.pluginDir ? options.pluginDir : ''
+  };
+
+  settings.urls = generateUrls(settings);
```

Comment:
Because they are dependent on other properties of the settings object. These values having additional logic to determine. suggestions?